src/core/QCheck2.ml

```diff
@@ -1397,104 +1543,109 @@ module Test = struct
   let fail_reportf m =
     let buf = Buffer.create 64 in
     Format.kfprintf
-      (fun out -> Format.fprintf out "@?"; fail_report (Buffer.contents buf))
-      (Format.formatter_of_buffer buf) m
+      (fun out ->
+        Format.fprintf out "@?" ;
+        fail_report (Buffer.contents buf))
+      (Format.formatter_of_buffer buf)
+      m
 
-  type 'a run_res =
-    | Run_ok
-    | Run_fail of string list
+  type 'a run_res = Run_ok | Run_fail of string list
 
   let run_law law x =
-    try
-      if law x then Run_ok else Run_fail []
-    with User_fail msg -> Run_fail [msg]
+    try if law x then Run_ok else Run_fail []
+    with User_fail msg -> Run_fail [ msg ]
 
   (* QCheck1-compatibility code *)
   exception Iter_exit
+
   let iter_find_map p iter =
     let r = ref None in
-    (try iter (fun x -> match p x with Some _ as y -> r := y; raise Iter_exit | None -> ())
-     with Iter_exit -> ()
-    );
+    (try
+       iter (fun x ->
+           match p x with
+           | Some _ as y ->
+               r := y ;
+               raise Iter_exit
+           | None -> ())
+     with Iter_exit -> ()) ;
     !r
 
   (* try to shrink counter-ex [i] into a smaller one. Returns
      shrinked value and number of steps *)
-  let shrink st (i_tree : 'a Tree.t) (r : res_or_exn) m : 'a * res_or_exn * string list * int =
-    let is_err = match r with
-      | Shrink_exn _ -> true | _ -> false
-    in
+  let shrink st (i_tree : 'a Tree.t) (r : res_or_exn) m :
+      'a * res_or_exn * string list * int =
+    let is_err = match r with Shrink_exn _ -> true | _ -> false in
     let rec shrink_ st i_tree r m ~steps =
-      let Tree.Tree (i, shrinks) = i_tree in
-      st.handler st.test.name st.test (Shrunk (steps, i));
+      let (Tree.Tree (i, shrinks)) = i_tree in
+      st.handler st.test.name st.test (Shrunk (steps, i)) ;
       let count = ref 0 in
-      let i' = match st.test.qcheck1_shrink with
-        | Some f -> (* QCheck1-compatibility, copied almost verbatim from QCheck.ml old code *)
-           iter_find_map
-             (fun x ->
-               (* let Tree.Tree (x, _) = x_tree in *)
-               try
-                 incr count;
-                 st.handler st.test.name st.test (Shrinking (steps, !count, x));
-                 begin match run_law st.test.law x with
-                 | Run_fail m when not is_err -> Some (Tree.pure x, Shrink_fail, m)
-                 | _ -> None
-                 end
```

Comment:
Looks like not every `begin end` is preserved.

Reply:
From what I understand, if a `begin end` statement is already in a group it is not preserved.

Reply:
back to the beginning then… :/

Reply:
I wish ocamlformat had a mode where it would just *not* touch parentheses. Let the programmer decide.